src/main/java/health/ere/ps/service/common/security/SecretsManagerService.java

```diff
@@ -142,52 +145,112 @@ public KeyStore initializeTrustStoreFromFile(String trustStoreFilePath,
+                                             String trustStoreFilePath,
+                                             String trustStorePassword,
+                                             boolean verifyHostnames,
                                              BindingProvider bp)
             throws SecretsManagerException {
-        try(FileInputStream fileInputStream = new FileInputStream(keyStoreFilePath)) {
-            SSLContext sc = createSSLContext(fileInputStream, keyStorePassword.toCharArray(),
-                sslContextType, keyStoreType);
+        try {
+            SSLContext sc = createSSLContext(keyStoreFilePath, keyStorePassword, sslContextType,
+                    trustStoreFilePath, trustStorePassword);
 
-            bp.getRequestContext().put("com.sun.xml.ws.transport.https.client.SSLSocketFactory",
-                    sc.getSocketFactory());
+            bp.getRequestContext().put(JAXWSProperties.SSL_SOCKET_FACTORY, sc.getSocketFactory());
 
+            if (!verifyHostnames) {
+                bp.getRequestContext().put(JAXWSProperties.HOSTNAME_VERIFIER, new SSLUtilities.FakeHostnameVerifier());
+            }
         } catch (IOException e) {
             // throw new SecretsManagerException("SSL transport configuration error.", e);
-            log.log(Level.SEVERE, "Could not configure SecretsManagerService", e);
+            LOG.log(Level.SEVERE, "Could not configure SecretsManagerService", e);
         }
     }
 
-    public SSLContext createSSLContext(InputStream keyStoreInputStream, char[] keyStorePassword,
-                                    SslContextType sslContextType, KeyStoreType keyStoreType)
+    /**
+     * Create a SSLContext with optional clientcertificate and optional trust store.
+     * <p>
+     * ATTENTION: if the trust store inputstream is null, all server certificates are accepted!
+     *
+     * @param keyStoreInputStream   inputstream with keystore content; if the stream is null, no client certificate will be used
+     * @param keyStorePassword      password to access the client certificate
+     * @param sslContextType        type of ssl context; should be TLS
+     * @param keyStoreType          type of the keystore with the client certificate
+     * @param trustStoreInputStream inputstream with trust store content; ATTENTION: if the stream is null, all server certificates are accepted without any validation
+     * @param trustStorePassword    password to access the trust store
+     * @param trustStoreType        type of the trust store with the server certificate
+     * @return SSLContext
+     * @throws SecretsManagerException will be thrown in case of invalid parameters
+     */
+    private SSLContext createSSLContext(InputStream keyStoreInputStream, char[] keyStorePassword,
+                                    SslContextType sslContextType, KeyStoreType keyStoreType,
+                                       InputStream trustStoreInputStream, char[] trustStorePassword,
```

Comment:
I think it is cleaner to use one capsuled record/data here for the trustStore. This record should have non nullable properties.

The parameters `InputStream trustStoreInputStream, char[] trustStorePassword, KeyStoreType trustStoreType) only have two states: all 3 are set or all 3 are null.